stdlib/socket/from.go

```diff
@@ -0,0 +1,201 @@
+// This source gets input from a socket connection and produces tables given a decoder.
+// This is a good candidate for streaming use cases. For now, it produces a single table for everything
+// that it receives from the start to the end of the connection.
+package socket
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/csv"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/line"
+	"github.com/influxdata/flux/plan"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+	"github.com/pkg/errors"
+)
+
+const FromSocketKind = "fromSocket"
+
+type FromSocketOpSpec struct {
+	URL     string `json:"url"`
+	Decoder string `json:"decoder"`
+}
+
+func init() {
+	fromSocketSignature := semantic.FunctionPolySignature{
+		Parameters: map[string]semantic.PolyType{
+			"url":     semantic.String,
+			"decoder": semantic.String,
+		},
+		Required: semantic.LabelSet{"url"},
+		Return:   flux.TableObjectType,
+	}
+
+	flux.RegisterPackageValue("socket", "from", flux.FunctionValue(FromSocketKind, createFromSocketOpSpec, fromSocketSignature))
+	flux.RegisterOpSpec(FromSocketKind, newFromSocketOp)
+	plan.RegisterProcedureSpec(FromSocketKind, newFromSocketProcedure, FromSocketKind)
+	execute.RegisterSource(FromSocketKind, createFromSocketSource)
+}
+
+type administrationTimeProvider struct {
+	a execute.Administration
+}
+
+func (a *administrationTimeProvider) GetCurrentTime() values.Time {
+	return a.a.ResolveTime(flux.Time{
+		IsRelative: false,
+		Absolute:   time.Now(),
+	})
```

Comment:
Why are we resolving times through the administration?

Reply:
I don't know actually. I thought that was the way to go for preserving the `now` specified by the user and so on.
The fact you are asking makes me think there is a much easier way 😅

Reply:
Using a.a.ResolveTime only matters if IsRelative is true. In this case it makes it look like it doing something useful when in reality its not. Lets just have the code return time.Now() converted to a values.Time. That way its simple and clear that no real resolving is happening.

Reply:
👍